public/systems_monitoring_graphs.js

```diff
@@ -0,0 +1,576 @@
+                    areaChart.select(".brush").call(brush.move, null) // This remove the grey brush area as soon as the selection has been done
+                }
+
+                // Update axis and area position
+                xAxis.transition().duration(1000).call(d3.axisBottom(x).tickFormat(dayTimeFormat))
+                // Rotate X axis ticks
+                xAxis.selectAll("text")
+                    .style("text-anchor", "end")
+                    .attr("dx", "-.8em")
+                    .attr("dy", ".15em")
+                    .attr("transform", "rotate(-65)");
+
+                areaChart
+                    .selectAll("path")
+                    .transition().duration(1000)
+                    .attr("d", area)
+            }
+
+            // What to do when one group is hovered
+            let highlight = function(d){
+                console.log(d)
+                // reduce opacity of all groups
+                d3.selectAll(".diskArea").style("opacity", .1)
+                // expect the one that is hovered
+                d3.select("."+d).style("opacity", 1)
+            }
+
+            // And when it is not hovered anymore
+            let noHighlight = function(d){
+                d3.selectAll(".diskArea").style("opacity", 1)
+            }
+
+            // Add legend for each name.
+            let size = 20
+            svg.selectAll("myrect")
+                .data(diskKeys)
+                .enter()
+                .append("rect")
+                    .attr("x", Width + 10)
+                    .attr("y", (d,i) => 10 + i*(size+5)) // 10 is where the first dot appears. 25 is the distance between dots
+                    .attr("width", size)
+                    .attr("height", size)
+                    .style("fill", d => color(d))
+                    .on("mouseover", highlight)
+                    .on("mouseleave", noHighlight)
+
+            // Add legend labels for each name.
+            svg.selectAll("mylabels")
+                .data(diskMetrics)
+                .enter()
+                .append("text")
+                    .attr("x", Width + 10 + size*1.2)
+                    .attr("y", (d,i) => 10 + i*(size+5) + (size/2)) // 10 is where the first dot appears. 25 is the distance between dots
+                    .style("fill", d => color(d))
+                    .text(d => d)
+                    .attr("text-anchor", "left")
+                    .style("alignment-baseline", "middle")
+                    .on("mouseover", highlight)
+                    .on("mouseleave", noHighlight)
+
+            // Disk usage graph title
+            svg.append("text")
+                .attr("x", Width / 2)
+                .attr("y", 0 - Margin.top / 2)
+                .attr("text-anchor", "middle")
+                .style("font-size", "20px")
+                .style("text-decoration", "bold")
+                .text("Disk Usage");
+
+        }
+
+        function plotMemoryMetrics(data) {
```

Comment:
Apart from disc usage; We should just plot used metric only for both memory and CPU utilization. Free disc space varies every time we delete files, but total memory & CPU metrics are constant so we can visually estimate the amount available.

Reply:
I'll just remove them from the stack, therefore after refactoring I will just have one function that can plot all the three graphs using different dataset